D-Astudillo-ASC/innovators-covid-project
# Added Route API's allowing for CRUD operations on database

# Description



Added an exam and route controller for route logic and CRUD commands. Added patient and exam controllers to
to route GET, PUT, POST, DELETE commands when accessing server-side. Also made edits to model and server side index.js. 

## Related Issues



Note that for exams when updating an exam the url does not update. So to update you must first delete that exam and create a new one with the new information you want so that the image url link can be updated.

## How to test




To test api use an api tester like postman or insomnia. In the patient and exam router files you will see how to access each command. The api url should be localhost:3000/api/(what the route command says in router files).
Example of this is localhost:3000/api/exams returns all exams in json format. Your local mongodb database must be named MedicalData and have two collections named 'patients' & 'exams'. Image below is how to get specific exam or patient. You must use the ID mongo creates for each patient or exam, not the patient id when conducting searches. Use patient & exam csv files to get real patient ID's. This is important as that is how the backend builds the link for the url image. Putting in the correct exam_id and xray_type is important for building the proper image url. Refer to patient and exam csv. Message me on slack if you're still confused, it was confusing for more at first as well.

![image](https://user-images.githubusercontent.com/68299334/152665249-62b6a8e4-2fe2-4fea-8b2d-7379db59755f.png)
